build_tools/circle/build_doc.sh

```diff
@@ -51,5 +52,11 @@ source activate testenv
 # Build and install scikit-learn in dev mode
 python setup.py develop
 
+if [[ "$CIRCLE_BRANCH" =~ ^master$|^[0-9]+\.[0-9]+\.X$ ]]
```

Comment:
> Hm but it would allow us to build pdf on PR branches if we want to.

I assume we wanted to mirror the behaviour of #7671, i.e. only build the pdf doc on pushes into master or into release branches and never inside a PR.